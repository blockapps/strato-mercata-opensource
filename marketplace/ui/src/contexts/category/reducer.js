import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createCategory:
      return {
        ...state,
        isCreateCategorySubmitting: true,
      };
    case actionDescriptors.createCategorySuccessful:
      return {
        ...state,
        category: action.payload,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.createCategoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.fetchCategory:
      return {
        ...state,
        iscategorysLoading: true,
      };
    case actionDescriptors.fetchCategorySuccessful:
      return {
        ...state,
        categorys: action.payload,
        iscategorysLoading: false,
      };
    case actionDescriptors.fetchCategoryFailed:
      return {
        ...state,
        error: action.error,
        iscategorysLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
