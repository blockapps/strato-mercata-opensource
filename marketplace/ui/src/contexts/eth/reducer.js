import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchETHSTAddress:
      return {
        ...state,
      };
    case actionDescriptors.fetchETHSTAddressSuccessful:
      return {
        ...state,
        ethstAddress: action.payload,
      };
    case actionDescriptors.fetchETHSTAddressFailed:
      return {
        ...state,
      };
    case actionDescriptors.addHash:
      return {
        ...state,
        isAddingHash: true,
      };
    case actionDescriptors.addHashSuccessful:
      return {
        ...state,
        isAddingHash: false,
      };
    case actionDescriptors.addHashFailed:
      return {
        ...state,
        isAddingHash: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
