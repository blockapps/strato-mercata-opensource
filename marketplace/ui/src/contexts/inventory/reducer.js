import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createInventory:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createInventoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchInventory:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUser:
      return {
        ...state,
        isUserInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventoryForUserSuccessful:
      return {
        ...state,
        userInventories: action.payload.data,
        userInventoriesTotal: action.payload.count,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUserFailed:
      return {
        ...state,
        error: action.error,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearch:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySearchSuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearchFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.updateInventory:
      return {
        ...state,
        isinventoryUpdating: true,
      };
    case actionDescriptors.updateInventorySuccessful:
      return {
        ...state,
        inventoryUpdateObject: action.payload,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateInventoryFailed:
      return {
        ...state,
        error: action.error,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateSale:
      return {
        ...state,
        issaleUpdating: true,
      };
    case actionDescriptors.updateSaleSuccessful:
      return {
        ...state,
        saleUpdateObject: action.payload,
        issaleUpdating: false,
      };
    case actionDescriptors.updateSaleFailed:
      return {
        ...state,
        error: action.error,
        issaleUpdating: false,
      };
    case actionDescriptors.listInventory:
      return {
        ...state,
        isListing: true,
      };
    case actionDescriptors.listInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isListing: false,
      };
    case actionDescriptors.listInventoryFailed:
      return {
        ...state,
        error: action.error,
        isListing: false,
      };
    case actionDescriptors.unlistInventory:
      return {
        ...state,
        isUnlisting: true,
      };
    case actionDescriptors.unlistInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isUnlisting: false,
      };
    case actionDescriptors.unlistInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnlisting: false,
      };

     // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.stakeInventory:
      return {
        ...state,
        isStaking: true
      };
    case actionDescriptors.stakeInventorySuccessful:
      return {
        ...state,
        isStaking: false
      };
    case actionDescriptors.stakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isStaking: false
      };
    
    case actionDescriptors.unstakeInventory:
      return {
        ...state,
        isUnstaking: true
      };
    case actionDescriptors.unstakeInventorySuccessful:
      return {
        ...state,
        isUnstaking: false
      };
    case actionDescriptors.unstakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnstaking: false
      };
    
    case actionDescriptors.borrow:
      return {
        ...state,
        isBorrowing: true
      };
    case actionDescriptors.borrowSuccessful:
      return {
        ...state,
        isBorrowing: false
      };
    case actionDescriptors.borrowFailed:
      return {
        ...state,
        error: action.error,
        isBorrowing: false
      };
    
    case actionDescriptors.repay:
      return {
        ...state,
        isRepaying: true
      };
    case actionDescriptors.repaySuccessful:
      return {
        ...state,
        isRepaying: false
      };
    case actionDescriptors.repayFailed:
        return {
          ...state,
          error: action.error,
          isRepaying: false
        };

    case actionDescriptors.getReserve:
        return {
          ...state,
          isReserveLoading: true
        };
    case actionDescriptors.getReserveSuccessful:
        return {
          ...state,
          reserve: action.payload,
          isReserveLoading: false
        };
    case actionDescriptors.getReserveFailed:
        return {
          ...state,
          error: action.error,
          isReserveLoading: false
        };
    
    case actionDescriptors.getAllReserve:
        return {
          ...state,
          isReservesLoading: true
        };
    case actionDescriptors.getAllReserveSuccessful:
        return {
          ...state,
          reserves: action.payload,
          isReservesLoading: false
        };
    case actionDescriptors.getAllReserveFailed:
        return {
          ...state,
          error: action.error,
          isReservesLoading: false
        };
    
    case actionDescriptors.getEscrowForAsset:
        return {
          ...state,
          isEscrowLoading: true
        };
    case actionDescriptors.getEscrowForAssetSuccessful:
        return {
          ...state,
          escrow: action.payload,
          isEscrowLoading: false
        };
    case actionDescriptors.getEscrowForAssetFailed:
        return {
          ...state,
          error: action.error,
          isEscrowLoading: false
        };

    case actionDescriptors.getUserCataRewards:
        return {
          ...state,
          isUserCataRewardsLoading: true
        };
    case actionDescriptors.getUserCataRewardsSuccessful:
        return {
          ...state,
          totalCataReward: action.payload.totalCataReward,
          dailyCataReward: action.payload.dailyCataReward,
          isUserCataRewardsLoading: false
        };
    case actionDescriptors.getUserCataRewardsFailed:
        return {
          ...state,
          error: action.error,
          isUserCataRewardsLoading: false
        };
    
    case actionDescriptors.getOracle:
          return {
            ...state,
            isFetchingOracle: true
          };
    case actionDescriptors.getOracleSuccessful:
          return {
            ...state,
            oracle: action.payload,
            isFetchingOracle: false
          };
    case actionDescriptors.getOracleFailed:
          return {
            ...state,
            error: action.error,
            isFetchingOracle: false
          };    
    // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.resellInventory:
      return {
        ...state,
        isReselling: true,
      };
    case actionDescriptors.resellInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isReselling: false,
      };
    case actionDescriptors.resellInventoryFailed:
      return {
        ...state,
        error: action.error,
        isReselling: false,
      };
    case actionDescriptors.fetchSupportedTokens:
      return {
        ...state,
        isFetchingTokens: true,
      };
    case actionDescriptors.fetchSupportedTokensSuccessful:
      return {
        ...state,
        supportedTokens: action.payload,
        isFetchingTokens: false,
      };
    case actionDescriptors.fetchSupportedTokensFailed:
      return {
        ...state,
        error: action.error,
        isFetchingTokens: false,
      };
    case actionDescriptors.bridgeInventory:
      return {
        ...state,
        isBridging: true,
      };
    case actionDescriptors.bridgeInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isBridging: false,
      };
    case actionDescriptors.bridgeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isBridging: false,
      };
    case actionDescriptors.transferInventory:
      return {
        ...state,
        isTransferring: true,
      };
    case actionDescriptors.transferInventorySuccessful:
      return {
        ...state,
        isTransferring: false,
      };
    case actionDescriptors.transferInventoryFailed:
      return {
        ...state,
        error: action.error,
        isTransferring: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistory:
      return {
        ...state,
        isInventoryOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchInventoryOwnershipHistorySuccessful:
      return {
        ...state,
        inventoryOwnershipHistory: action.payload,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryDetail:
      return {
        ...state,
        isInventoryDetailsLoading: true,
      };
    case actionDescriptors.fetchInventoryDetailSuccessful:
      return {
        ...state,
        inventoryDetails: action.payload,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.fetchInventoryDetailFailed:
      return {
        ...state,
        error: action.error,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.uploadImage:
      return {
        ...state,
        isUploadImageSubmitting: true,
      };
    case actionDescriptors.uploadImageSuccessful:
      return {
        ...state,
        uploadedImg: action.payload,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.uploadImageFailed:
      return {
        ...state,
        error: action.error,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.createItem:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createItemSuccessful:
      return {
        ...state,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createItemFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchPriceHistory:
      return {
        ...state,
        isFetchingPriceHistory: true,
      };
    case actionDescriptors.fetchPriceHistorySuccessful:
      return {
        ...state,
        isFetchingPriceHistory: false,
        priceHistory: action.payload,
      };
    case actionDescriptors.fetchPriceHistoryFailed:
      return {
        ...state,
        isFetchingPriceHistory: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
