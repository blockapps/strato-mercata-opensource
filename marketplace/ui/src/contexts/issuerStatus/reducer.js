import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.requestReview:
      return {
        ...state,
        requestingReview: true,
      };
    case actionDescriptors.requestReviewSuccessful:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.requestReviewFailed:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.authorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.authorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.authorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.deauthorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.modifyAdmin:
      return {
        ...state,
        changingAdminStatus: true,
      };
    case actionDescriptors.modifyAdminSuccessful:
      return {
        ...state,
        changingAdminStatus: false,
      };
    case actionDescriptors.modifyAdminFailed:
      return {
        ...state,
        changingAdminStatus: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
