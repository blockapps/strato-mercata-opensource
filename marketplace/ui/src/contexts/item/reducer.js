import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchItem:
      return {
        ...state,
        isItemsLoading: true,
      };
    case actionDescriptors.fetchItemSuccessful:
      return {
        ...state,
        items: action.payload,
        isItemsLoading: false,
      };
    case actionDescriptors.fetchItemFailed:
      return {
        ...state,
        error: action.error,
        isItemsLoading: false,
      };
    case actionDescriptors.fetchSerialNumbers:
      return {
        ...state,
        isSerialNumbersLoading: true,
      };
    case actionDescriptors.fetchSerialNumbersSuccessful:
      return {
        ...state,
        serialNumbers: action.payload,
        isSerialNumbersLoading: false,
      };
    case actionDescriptors.fetchSerialNumbersFailed:
      return {
        ...state,
        error: action.error,
        isSerialNumbersLoading: false,
      };
    case actionDescriptors.fetchItemOwnershipHistory:
      return {
        ...state,
        isOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchItemOwnershipHistorySuccessful:
      return {
        ...state,
        ownershipHistory: action.payload,
        isOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchItemOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemRawMaterials:
      return {
        ...state,
        isRawMaterialsLoading: true,
      };
    case actionDescriptors.fetchItemRawMaterialsSuccessful:
      return {
        ...state,
        rawMaterials: action.payload,
        isRawMaterialsLoading: false,
      };
    case actionDescriptors.fetchItemRawMaterialsFailed:
      return {
        ...state,
        error: action.error,
        isRawMaterialsLoading: false,
      };
    case actionDescriptors.setActualRawMaterials:
      return {
        ...state,
        actualRawMaterials: action.payload,
      };
    case actionDescriptors.transferOwnership:
      return {
        ...state,
        isTransferringItems: true,
      };
    case actionDescriptors.transferOwnershipSuccessful:
      return {
        ...state,
        isTransferringItems: false,
      };
    case actionDescriptors.transferOwnershipFailed:
      return {
        ...state,
        error: action.error,
        isTransferringItems: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
