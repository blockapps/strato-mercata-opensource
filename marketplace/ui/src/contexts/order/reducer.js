import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createOrder:
      return {
        ...state,
        isCreateOrderSubmitting: true,
      };
    case actionDescriptors.createOrderSuccessful:
      return {
        ...state,
        order: action.payload,
        isCreateOrderSubmitting: false,
      };
    case actionDescriptors.createOrderFailed:
      return {
        ...state,
        error: action.error,
        isCreateOrderSubmitting: false,
      };
    case actionDescriptors.cancelSale:
      return {
        ...state,
        isorderDetailsLoading: true,
      };
    case actionDescriptors.cancelSaleSuccessful:
      return {
        ...state,
        isorderDetailsLoading: false,
      };
    case actionDescriptors.cancelSaleFailed:
      return {
        ...state,
        isorderDetailsLoading: false,
        error: action.error,
      };
    case actionDescriptors.createPayment:
      return {
        ...state,
        isCreatePaymentSubmitting: true,
      };
    case actionDescriptors.createPaymentSuccessful:
      return {
        ...state,
        payment: action.payload,
        isCreatePaymentSubmitting: false,
      };
    case actionDescriptors.createPaymentFailed:
      return {
        ...state,
        error: action.error,
        isCreatePaymentSubmitting: false,
      };
    case actionDescriptors.createOrderLineItem:
      return {
        ...state,
        isCreateOrderLineItem: true,
      };
    case actionDescriptors.createOrderLineItemSuccessful:
      return {
        ...state,
        item: action.payload,
        isCreateOrderLineItem: false,
      };
    case actionDescriptors.createOrderLineItemFailed:
      return {
        ...state,
        error: action.error,
        isCreateOrderLineItem: false,
      };
    case actionDescriptors.fetchOrder:
      return {
        ...state,
        isordersLoading: true,
      };
    case actionDescriptors.fetchOrderSuccessful:
      return {
        ...state,
        orders: action.payload.orders,
        orderBoughtTotal: action.payload.total,
        isordersLoading: false,
      };
    case actionDescriptors.fetchOrderFailed:
      return {
        ...state,
        error: action.error,
        isordersLoading: false,
      };
    case actionDescriptors.fetchAllOrders:
      return {
        ...state,
        isAllOrdersLoading: true,
      };
    case actionDescriptors.fetchAllOrdersSuccessful:
      return {
        ...state,
        allOrders: action.payload,
        isAllOrdersLoading: false,
      };
    case actionDescriptors.fetchAllOrdersFailed:
      return {
        ...state,
        error: action.error,
        isAllOrdersLoading: false,
      };
    case actionDescriptors.fetchSaleQuantity:
      return {
        ...state,
        saleQuantityLoading: true,
      };
    case actionDescriptors.fetchSaleQuantitySuccessful:
      return {
        ...state,
        saleQuantity: action.payload,
        saleQuantityLoading: false,
      };
    case actionDescriptors.fetchSaleQuantityFailed:
      return {
        ...state,
        error: action.error,
        saleQuantityLoading: false,
      };
    case actionDescriptors.fetchOrderSold:
      return {
        ...state,
        isordersSoldLoading: true,
      };
    case actionDescriptors.fetchOrderSoldSuccessful:
      return {
        ...state,
        ordersSold: action.payload.orders,
        orderSoldTotal: action.payload.total,
        isordersSoldLoading: false,
      };
    case actionDescriptors.fetchOrderSoldFailed:
      return {
        ...state,
        error: action.error,
        isordersSoldLoading: false,
      };
    case actionDescriptors.fetchOrderDetails:
      return {
        ...state,
        isorderDetailsLoading: true,
      };
    case actionDescriptors.fetchOrderDetailsSuccessful:
      return {
        ...state,
        orderDetails: action.payload,
        isorderDetailsLoading: false,
      };
    case actionDescriptors.fetchOrderDetailsFailed:
      return {
        ...state,
        error: action.error,
        isorderDetailsLoading: false,
      };
    case actionDescriptors.fetchOrderLineItemDetails:
      return {
        ...state,
        isOrderLineDetailsLoading: true,
      };
    case actionDescriptors.fetchOrderLineItemDetailsSuccessful:
      return {
        ...state,
        orderLineDetails: action.payload,
        isOrderLineDetailsLoading: false,
      };
    case actionDescriptors.fetchOrderLineItemDetailsFailed:
      return {
        ...state,
        error: action.error,
        isOrderLineDetailsLoading: false,
      };
    case actionDescriptors.executeSale:
      return {
        ...state,
        isCreateOrderSubmitting: true,
      };
    case actionDescriptors.executeSaleSuccessful:
      return {
        ...state,
        isCreateOrderSubmitting: false,
      };
    case actionDescriptors.executeSaleFailed:
      return {
        ...state,
        error: action.error,
        isCreateOrderSubmitting: false,
      };
    case actionDescriptors.waitForOrderEvent:
      return {
        ...state,
        isOrderEventLoading: true,
      };
    case actionDescriptors.waitForOrderEventSuccessful:
      return {
        ...state,
        isOrderEventLoading: false,
      };
    case actionDescriptors.waitForOrderEventFailed:
      return {
        ...state,
        error: action.error,
        isOrderEventLoading: false,
      };
    case actionDescriptors.updateOrderComment:
      return {
        ...state,
        isUpdatingOrderComment: true,
      };
    case actionDescriptors.updateOrderCommentSuccessful:
      return {
        ...state,
        isUpdatingOrderComment: false,
      };
    case actionDescriptors.updateOrderCommentFailed:
      return {
        ...state,
        error: action.error,
        isUpdatingOrderComment: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
