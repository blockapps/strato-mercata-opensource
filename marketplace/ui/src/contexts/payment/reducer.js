import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.getPaymentServices:
      return {
        ...state,
        arePaymentServicesLoading: true,
      };
    case actionDescriptors.getPaymentServicesSuccessful:
      return {
        ...state,
        paymentServices: action.payload.data,
        paymentServicesTotal: action.payload.count,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getPaymentServicesFailed:
      return {
        ...state,
        error: action.error,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getNotOnboarded:
      return {
        ...state,
        areNotOnboardedLoading: true,
      };
    case actionDescriptors.getNotOnboardedSuccessful:
      return {
        ...state,
        notOnboarded: action.payload.data,
        notOnboardedTotal: action.payload.count,
        areNotOnboardedLoading: false,
      };
    case actionDescriptors.getNotOnboardedFailed:
      return {
        ...state,
        error: action.error,
        areNotOnboardedLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
