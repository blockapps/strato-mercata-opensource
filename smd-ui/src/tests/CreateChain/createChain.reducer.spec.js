import reducer from '../../components/CreateChain/createChain.reducer';
import {
  openCreateChainOverlay,
  closeCreateChainOverlay,
  openAddMemberModal,
  closeAddMemberModal,
  resetError,
  createChain,
  createChainSuccess,
  createChainFailure,
  compileChainContractFailure,
  compileChainContractSuccess,
  resetContract
} from '../../components/CreateChain/createChain.actions';
import { xabiMock } from './createChainMock';

describe('CreateChain: reducer', () => {

  // INITIAL_STATE
  test('set initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  describe('create chain modal', () => {

    // OPEN_OVERLAY
    test('open overlay', () => {
      let initialState = {
        isAddMemberModalOpen: false,
        isOpen: false,
        spinning: false,
        key: null,
        error: null,
        abi: null
      };

      const action = openCreateChainOverlay();
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // CLOSE_OVERLAY
    test('close overlay', () => {
      let initialState = {
        isAddMemberModalOpen: false,
        isOpen: false,
        spinning: false,
        key: null,
        error: null
      };

      const action = closeCreateChainOverlay();
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

  });

  describe('add member modal', () => {

    // OPEN_ADD_MEMBER_MODAL
    test('open overlay', () => {
      const initialState = {
        isAddMemberModalOpen: false,
        isOpen: false,
        spinning: false,
        key: null,
        error: null
      };

      const action = openAddMemberModal();
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // CLOSE_ADD_MEMBER_MODAL
    test('close overlay', () => {
      const initialState = {
        isAddMemberModalOpen: true,
        isOpen: true,
        spinning: false,
        key: null,
        error: null
      };

      const action = closeAddMemberModal();
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

  });

  describe('create chain', () => {

    const payload = {
      label: 'airline cartel 9',
      members: [{ address: "f11b5c42f5b84efa07f6b0a32c3fc545ff509126", enode: "enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7â€¦4cac9f77166ad92a0@172.16.0.5:30303?discport=30303" }],
      balances: [{ balance: 500000000000000, address: "f11b5c42f5b84efa07f6b0a32c3fc545ff509126" }],
      src: `contract SimpleStorage {
        uint public storedData;
      }`,
      args: { addRule: "MajorityRules", removeRule: "MajorityRules" }
    }

    // CREATE_CHAIN_REQUEST
    test('request', () => {
      const initialState = {
        isAddMemberModalOpen: false,
        isOpen: false,
        spinning: false,
        key: null,
        error: null
      };

      const action = createChain(payload.label, payload.members, payload.balances, payload.src, payload.args);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // CREATE_CHAIN_SUCCESS
    test('success', () => {
      const initialState = {
        isAddMemberModalOpen: false,
        isOpen: true,
        spinning: true,
        key: null,
        error: null
      };

      const action = createChainSuccess('64885c49cdc6fe5f15975596115a120ec1e9a616e88a22e0be0457f373d75b73');
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // CREATE_CHAIN_FAILURE
    test('failure', () => {
      const initialState = {
        isAddMemberModalOpen: false,
        isOpen: true,
        spinning: true,
        key: null,
        error: null
      };

      const action = createChainFailure('error');
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

  })

  describe('compile chain contract', () => {

    // COMPILE_CHAIN_CONTRACT_SUCCESS
    test('success', () => {
      const initialState = {
        isAddMemberModalOpen: false,
        isOpen: false,
        spinning: false,
        key: null,
        error: null,
        abi: null
      };

      const action = compileChainContractSuccess(xabiMock);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // COMPILE_CHAIN_CONTRACT_FAILURE
    test('failure', () => {
      const initialState = {
        isAddMemberModalOpen: false,
        isOpen: true,
        spinning: true,
        key: null,
        error: null,
        abi: null
      };

      const action = compileChainContractFailure('error');
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

  })

  test('reset error', () => {
    const initialState = {
      isAddMemberModalOpen: true,
      isOpen: true,
      spinning: false,
      key: null,
      error: 'error'
    };

    const action = resetError();
    expect(reducer(initialState, action)).toMatchSnapshot();
  });

  test('reset contract', () => {
    const initialState = {
      isAddMemberModalOpen: true,
      isOpen: true,
      spinning: false,
      key: null,
      error: null,
      abi: 'contract test {}'
    };

    const action = resetContract();
    expect(reducer(initialState, action)).toMatchSnapshot();
  });

});