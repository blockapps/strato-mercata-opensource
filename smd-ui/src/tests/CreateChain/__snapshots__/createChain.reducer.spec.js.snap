// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateChain: reducer add member modal close overlay 1`] = `
Object {
  "error": null,
  "isAddMemberModalOpen": false,
  "isOpen": true,
  "key": null,
  "spinning": false,
}
`;

exports[`CreateChain: reducer add member modal open overlay 1`] = `
Object {
  "error": null,
  "isAddMemberModalOpen": true,
  "isOpen": false,
  "key": null,
  "spinning": false,
}
`;

exports[`CreateChain: reducer compile chain contract failure 1`] = `
Object {
  "abi": null,
  "error": "error",
  "isAddMemberModalOpen": false,
  "isOpen": true,
  "key": null,
  "spinning": true,
}
`;

exports[`CreateChain: reducer compile chain contract success 1`] = `
Object {
  "abi": Object {
    "src": Object {
      "Governance": Object {
        "constr": Object {},
        "events": Object {},
        "funcs": Object {},
        "isLibrary": false,
        "modifiers": Object {},
        "types": Object {},
        "using": Object {},
        "vars": Object {
          "addRule": Object {
            "atBytes": 0,
            "constant": true,
            "dynamic": true,
            "initialValue": "\\"MajorityRules\\"",
            "public": null,
            "type": "String",
          },
          "removeRule": Object {
            "atBytes": 0,
            "constant": true,
            "dynamic": true,
            "initialValue": "\\"MajorityRules\\"",
            "public": null,
            "type": "String",
          },
          "temp": Object {
            "atBytes": 0,
            "constant": null,
            "dynamic": true,
            "initialValue": null,
            "public": null,
            "type": "String",
          },
        },
      },
    },
  },
  "contractName": "Governance",
  "error": null,
  "isAddMemberModalOpen": false,
  "isOpen": false,
  "key": null,
  "spinning": false,
}
`;

exports[`CreateChain: reducer create chain failure 1`] = `
Object {
  "error": "error",
  "isOpen": false,
  "spinning": false,
}
`;

exports[`CreateChain: reducer create chain modal close overlay 1`] = `
Object {
  "isOpen": false,
}
`;

exports[`CreateChain: reducer create chain modal open overlay 1`] = `
Object {
  "isOpen": true,
}
`;

exports[`CreateChain: reducer create chain request 1`] = `
Object {
  "isOpen": true,
  "spinning": true,
}
`;

exports[`CreateChain: reducer create chain success 1`] = `
Object {
  "isOpen": false,
  "key": "64885c49cdc6fe5f15975596115a120ec1e9a616e88a22e0be0457f373d75b73",
  "spinning": false,
}
`;

exports[`CreateChain: reducer reset contract 1`] = `
Object {
  "abi": null,
  "error": null,
  "isAddMemberModalOpen": true,
  "isOpen": true,
  "key": null,
  "spinning": false,
}
`;

exports[`CreateChain: reducer reset error 1`] = `
Object {
  "error": null,
  "isAddMemberModalOpen": true,
  "isOpen": true,
  "key": null,
  "spinning": false,
}
`;

exports[`CreateChain: reducer set initial state 1`] = `
Object {
  "abi": null,
  "contractName": undefined,
  "error": null,
  "isAddIntegrationModalOpen": false,
  "isAddMemberModalOpen": false,
  "isOpen": false,
  "key": null,
  "spinning": false,
}
`;
